arbor/include/arbor/communication/remote.hpp

```diff
@@ -0,0 +1,182 @@
+#pragma once
+
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <cstdint>
+#include <exception>
+#include <variant>
+
+#include <mpi.h>
+
+namespace arb {
+namespace remote {
+
+// Magic protocol tag
+constexpr std::uint8_t ARB_REMOTE_MAGIC = 0xAB;
+
+// Remote protocol version
+constexpr std::uint8_t ARB_REMOTE_VERSION_MAJOR = 0x00;
+constexpr std::uint8_t ARB_REMOTE_VERSION_MINOR = 0x01;
+constexpr std::uint8_t ARB_REMOTE_VERSION_PATCH = 0x00;
+
+// Message buffer length
+constexpr std::size_t ARB_REMOTE_MESSAGE_LENGTH = 1024;
```

Comment:
can't this be inferred from the variant type?

Reply:
Could be done, true. Idea here is to potentially interact with less sophisticated languages, think C or Python, that only have byte buffers.  Whomever also might elect not to use this header.
Having a fixed-size buffer with some headroom seemed sensible under this light.

Reply:
You mean the rough equivalent to `static_assert(sizeof(std::variant<...> < 1024)`?

Reply:
yes, something like this, including the meta bits